pkg/controller/scheduler.go

```diff
@@ -670,15 +671,26 @@ func (c *Controller) runBlueGreen(canary *flaggerv1.Canary, canaryController can
 
 func (c *Controller) runAnalysis(canary *flaggerv1.Canary) bool {
 	// run external checks
+	var err error
+	targetIterations := strconv.Itoa(canary.GetAnalysis().Iterations)
+	currentIterations := strconv.Itoa(canary.Status.Iterations)
+
 	for _, webhook := range canary.GetAnalysis().Webhooks {
 		if webhook.Type == "" || webhook.Type == flaggerv1.RolloutHook {
-			err := CallWebhook(canary.Name, canary.Namespace, flaggerv1.CanaryPhaseProgressing, webhook)
-			if err != nil {
-				c.recordEventWarningf(canary, "Halt %s.%s advancement external check %s failed %v",
-					canary.Name, canary.Namespace, webhook.Name, err)
-				return false
+			c.recordEventInfof(canary, "running rollout webhooks targetIterations: %s , currentIterations: %s", targetIterations, currentIterations)
+			if canary.GetAnalysis().StepWeight > 0 || len(canary.GetAnalysis().StepWeights) > 0 {
+				err = CallWebhook(canary.Name, canary.Namespace, flaggerv1.CanaryPhaseProgressing, webhook)
+				c.recordEventInfof(canary, "calling rollout webhook %s for canary", webhook.Name)
+			} else if canary.Status.Iterations == canary.GetAnalysis().Iterations || len(canary.GetAnalysis().Match) > 0 {
```

Comment:
The `canary.Status.Iterations == canary.GetAnalysis().Iterations` makes the webhooks run only once for B/G analysis right at the end, while we want to run it for every iteration.

Reply:
Why do we want it to run on every iteration? e.g, if we use the Rollout webhooks for external tests and run e2e on a new version (green), wouldn't it make sense to run it after traffic split and only once, otherwise we are testing the current version (blue)?